metrics-operator/controllers/metrics/controller.go

```diff
@@ -142,3 +170,111 @@ func (r *KeptnMetricReconciler) fetchProvider(ctx context.Context, namespacedMet
 	}
 	return provider, nil
 }
+
+func aggregateValues(stringSlice []string, aggFunc string) (string, error) {
+	floatSlice, err := stringSliceToFloatSlice(stringSlice)
+	if err != nil {
+		return "", err
+	}
+	var aggValue string
+	switch aggFunc {
+	case "max":
+		aggValue = fmt.Sprintf("%v", calculateMax(floatSlice))
+	case "min":
+		aggValue = fmt.Sprintf("%v", calculateMin(floatSlice))
+	case "median":
+		aggValue = fmt.Sprintf("%v", calculateMedian(floatSlice))
+	case "avg":
+		aggValue = fmt.Sprintf("%v", calculateAverage(floatSlice))
+	case "p90":
+		aggValue = fmt.Sprintf("%v", calculatePercentile(sort.Float64Slice(floatSlice), 0.90))
+	case "p95":
+		aggValue = fmt.Sprintf("%v", calculatePercentile(sort.Float64Slice(floatSlice), 0.95))
+	case "p99":
+		aggValue = fmt.Sprintf("%v", calculatePercentile(sort.Float64Slice(floatSlice), 0.99))
+	}
+	return aggValue, nil
+}
+
+func stringSliceToFloatSlice(strSlice []string) ([]float64, error) {
+	floatSlice := make([]float64, len(strSlice))
+
+	for i, str := range strSlice {
+		floatValue, err := strconv.ParseFloat(str, 64)
+		if err != nil {
+			return nil, err
+		}
+		floatSlice[i] = floatValue
+	}
+
+	return floatSlice, nil
+}
+
+func calculateMax(values []float64) float64 {
+	if len(values) == 0 {
+		return 0.0
+	}
+
+	max := values[0]
+	for _, value := range values {
+		if value > max {
+			max = value
+		}
+	}
+	return max
+}
+
+func calculateMin(values []float64) float64 {
+	if len(values) == 0 {
+		return 0.0
+	}
+
+	min := values[0]
+	for _, value := range values {
+		if value < min {
+			min = value
+		}
+	}
+	return min
+}
+
+func calculateMedian(values []float64) float64 {
+	if len(values) == 0 {
+		return 0.0
+	}
+
+	// Sort the values
+	sortedValues := make([]float64, len(values))
+	copy(sortedValues, values)
+	sort.Float64s(sortedValues)
+
+	// Calculate the median
+	middle := len(sortedValues) / 2
+	if len(sortedValues)%2 == 0 {
+		return (sortedValues[middle-1] + sortedValues[middle]) / 2
+	}
+	return sortedValues[middle]
+}
+
+func calculateAverage(values []float64) float64 {
+	sum := 0.0
+
+	for _, value := range values {
+		sum += value
+	}
+	if len(values) > 0 {
+		return sum / float64(len(values))
+	}
+
+	return 0.0
+}
+
+func calculatePercentile(values sort.Float64Slice, perc float64) float64 {
+	if len(values) == 0 {
+		return 0.0
+	}
+	// Calculate the index for the requested percentile
+	i := int(float64(len(values)) * perc / 100)
```

Comment:
i think here we need to use `math.Ceil`:
```suggestion
	i := math.Ceil(float64(len(values)) * perc / 100)
```